File: src/main/java/com/tibiawiki/domain/enums/Spawntype.java
package com.tibiawiki.domain.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import com.tibiawiki.domain.interfaces.Description;

public enum Spawntype implements Description {

    REGULAR("Regular"),
    RAID("Raid"),
    EVENT("Event"),
    UNIQUE("Unique"),
    TRIGGERED("Triggered"),
    UNBLOCKABLE("Unblockable"),
    EMPTY("");

    private String description;

    Spawntype(String description) {
        this.description = description;
    }

    @JsonValue
    public String getDescription() {
        return description;
    }
}

File: src/main/kotlin/com/tibiawiki/domain/enums/Spawntype.kt
package com.tibiawiki.domain.enums

import com.fasterxml.jackson.annotation.JsonValue
import com.tibiawiki.domain.interfaces.Description

enum class Spawntype(
    @get:JsonValue override val description: String
) : Description {
    REGULAR("Regular"),
    RAID("Raid"),
    EVENT("Event"),
    UNIQUE("Unique"),
    TRIGGERED("Triggered"),
    UNBLOCKABLE("Unblockable"),
    EMPTY("")
}
